### PR TITLE
toJSON and toString

### PR DESCRIPTION
**Pull request by DonutEspresso**
Implement these two methods for better stand alone formatting (see restify/node-restify#952).

Also fixes some inconsistencies in the stringified output with regards to having 'Error' at the end of the code.

### CODE REVIEW

**Comment by DonutEspresso** [comment]
`code` property now has a value of 'Error' for HttpError and RestError. Any subclass will have the name of the error, minus 'Error', i.e., GatewayTimeoutError has a code of GatewayTimeout. All `code` and `restCode` properties are now overridable. 

If all looks good, let's get this merged.

File: lib/index.js [hunk 1]
```diff
@@ -54,6 +54,14 @@ function makeConstructor(name, defaults) {
     assert.string(name, 'name');
     assert.optionalObject(defaults, 'defaults');
 
+    // code property doesn't have 'Error' in it. remove it.
+    var defaultCode = name.replace(new RegExp('Error$'), '');
```
Comment (micahr) [inline lib/index.js]: Probably should check for capital or lowercase Error

File: lib/index.js [hunk 2]
```diff
@@ -54,6 +54,14 @@ function makeConstructor(name, defaults) {
...
+    var d = _.assign({}, {
+        name: name,
+        code: (defaults && defaults.code) || defaultCode,
```
Comment (micahr) [inline lib/index.js]: `_.get(defaults, 'code', defaultCode)` looks a bit nicer :)

**Comment by micahr** [comment]
LGTM :shipit:

**Comment by DonutEspresso** [comment]
Addressed comments - final pass and I'll get it merged. @yunong FYI

**Comment by DonutEspresso** [comment]
Gonna merge this sometime today, I think this will require a major rev due to changes in the `code` and `restCode` properties.

**Pull request merged**